packages/xo-web/src/xo-app/proxies/deploy-proxy.js

```diff
@@ -228,8 +228,27 @@ const Modal = decorate([
   ),
 ])
 
-const deployProxy = ({ license, proxy }) => {
+const deployProxy = async ({ proxy } = {}) => {
+  const licenses = await getLicenses({ productType: 'xoproxy' })
   const isRedeployMode = proxy !== undefined
+
+  let license
+  if (isRedeployMode) {
+    license = licenses.find(
+      license =>
+        !(license.expires < Date.now()) &&
+        license.boundObjectId === proxy.vmUuid
+    )
+  }
+
+  // in case of deploying a proxy or when the associated proxy VM doesn't have a license
+  if (license === undefined) {
+    license = licenses.find(
+      license =>
+        !(license.expires < Date.now()) && license.boundObjectId === undefined
+    )
+  }
```

Comment:
What happens if no license has been found?

Reply:
There have a special display for this case

![image](https://user-images.githubusercontent.com/21563339/80706296-5f0db100-8ae8-11ea-8dbd-669a1c6f9146.png)